### PR TITLE
Add tests for file/folder filter

### PR DESCRIPTION
**Pull request by paurakhsharma**
## Description
Add webUI tests for file/folder filter.

## Related Issue
#844

## Motivation and Context
Add more tests

## How Has This Been Tested?
🤖

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
Find screenshots at https://minio.owncloud.com/phoenix/screenshots/2599/

**Comment by ownclouders** [comment]
Find screenshots at https://minio.owncloud.com/phoenix/screenshots/2603/

File: tests/acceptance/pageObjects/filesPage.js [hunk 1]
```diff
@@ -101,6 +100,50 @@ module.exports = {
...
+      this.api.element('css selector', filterCheckboxId, (response) => {
+        this.api.elementIdSelected(response.value.ELEMENT, (result) => {
+          if (enableOrDisable === 'disable' && result.value === true) {
+            // if the filter needs to be disabled but it is enabled
+            return this
+              .click(filterCheckbox)
+              .click('@filterListButton')
+          } else if (enableOrDisable === 'enable' && result.value === false) {
+            // if the filter needs to be enabled but it is disabled
+            return this
```
Comment (individual-it) [inline tests/acceptance/pageObjects/filesPage.js]: same as in the `if` block

**Comment by ownclouders** [comment]
Find screenshots at https://minio.owncloud.com/phoenix/screenshots/2613/

File: tests/acceptance/pageObjects/filesPage.js [hunk 2]
```diff
@@ -101,6 +100,50 @@ module.exports = {
...
+    toggleFilterFileOrFolder: async function (fileOrFolder, enableOrDisable) {
+      const filterSelector = {}
+      if (fileOrFolder === 'folder') {
+        filterSelector['filterLabel'] = '@filterFolderLabel'
+        filterSelector['filterCheckbox'] = '@filterFolderCheckbox'
+        filterSelector['filterCheckboxId'] = this.elements['filterFolderCheckbox']
+      } else if (fileOrFolder === 'file') {
+        filterSelector['filterLabel'] = '@filterFileLabel'
+        filterSelector['filterCheckbox'] = '@filterFileCheckbox'
+        filterSelector['filterCheckboxId'] = this.elements['filterFileCheckbox']
```
Comment (individual-it) [inline tests/acceptance/pageObjects/filesPage.js]: try to use only one type of addressing

File: tests/acceptance/pageObjects/filesPage.js [hunk 3]
```diff
@@ -101,6 +100,50 @@ module.exports = {
         .useXpath()
         .waitForElementNotPresent('@loadingIndicator')
         .waitForElementNotPresent(this.getFileRowSelectorByFileName(element))
+    },
+    toggleFilterFileOrFolder: async function (fileOrFolder, enableOrDisable) {
```
Comment (individual-it) [inline tests/acceptance/pageObjects/filesPage.js]: why async?

File: tests/acceptance/pageObjects/filesPage.js [hunk 4]
```diff
@@ -101,6 +100,50 @@ module.exports = {
...
+        .waitForElementVisible(filterSelector.filterLabel)
+
+      return this
+        .toogleCheckbox(
+          enableOrDisable,
+          filterSelector.filterCheckbox,
+          filterSelector.filterCheckboxId
+        )
+    },
+    toogleCheckbox: function (enableOrDisable, filterCheckbox, filterCheckboxId) {
```
Comment (individual-it) [inline tests/acceptance/pageObjects/filesPage.js]: maybe we have to place it in customCommands see http://nightwatchjs.org/guide#writing-custom-commands

File: tests/acceptance/features/webUIFiles/filter.feature [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+      | strängé नेपाली folder empty |
+      | strängé नेपाली folder       |
+
+  Scenario: files should not be listed in inside a folder if the files are filtered out
+    When the user disables file filter using the webUI
+    And the user opens folder "simple-folder" using the webUI
+    Then file "lorem.txt" should not be listed on the webUI
+    But folder "simple-empty-folder" should be listed on the webUI
+
+  Scenario: folder should not be listed in inside a folder if the folders are filtered out
```
Comment (individual-it) [inline tests/acceptance/features/webUIFiles/filter.feature]: s/in inside/inside

File: tests/acceptance/helpers/webUIAction.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  client.element('css selector', checkboxId, (response) => {
+    client.elementIdSelected(response.value.ELEMENT, (result) => {
+      const checked = result.value === true
+      if ((enableOrDisable === 'disable' && checked) ||
+        (enableOrDisable === 'enable' && !checked)) {
+        // if the checkbox needs to be unchecked but it is checked
+        // Or the checkbox needs to be checked but it is unchecked
+        return pageObject
+          .click(checkboxId)
+          .click('@filterListButton')
```
Comment (individual-it) [inline tests/acceptance/helpers/webUIAction.js]: the selector might be called something else if we use the command from an other place

File: tests/acceptance/pageObjects/filesPage.js [hunk 7]
```diff
@@ -119,6 +120,28 @@ module.exports = {
...
+        labelSelector = '@filterFileLabel'
+        checkboxId = this.elements['filterFileCheckbox']
+      } else {
+        throw new Error(`Expected 'file' or 'folder', ${fileOrFolder} given`)
+      }
+      return this
+        .waitForElementVisible('@filterListButton')
+        .click('@filterListButton')
+        .waitForElementVisible(labelSelector, () => {
+          webUIAction.toggleCheckbox(
```
Comment (individual-it) [inline tests/acceptance/pageObjects/filesPage.js]: we should be able to use
```suggestion
          .waitForElementVisible(labelSelector)
          .webUIAction.toggleCheckbox(enableOrDisable, checkboxId)
```

**Comment by ownclouders** [comment]
Find screenshots at https://minio.owncloud.com/phoenix/screenshots/2633/

**Pull request merged**